src/storage/nats_kv.rs

```diff
@@ -0,0 +1,106 @@
+//! Storage engine backed by NATS Kv
+//!
+//! This storage engine enables storing any WADM related information using NATS Kv as a backend
+
+use async_trait::async_trait;
+
+use async_nats::{
+    Client,
+    Error as NatsError,
+    Subscriber,
+};
+
+use semver::{Version, Prerelease, BuildMetadata};
+
+use super::{
+    Engine,
+    LatticeId,
+    LatticeStorage,
+    StorageEngine,
+    StorageEngineError,
+    EngineName,
+    StorageError,
+};
+
+use super::state::{
+    LatticeState,
+    LatticeStateBuilder,
+};
+
+const STORAGE_ENGINE_NAME: &'static str = "nats_kv";
+const STORAGE_ENGINE_VERSION: Version = Version {
+    major: 0,
+    minor: 1,
+    patch: 0,
+    pre: Prerelease::EMPTY,
+    build: BuildMetadata::EMPTY,
+};
+
+use crate::storage::Stores;
+
+#[derive(Debug)]
+struct NatsKvStorageEngine {
+    /// URL to the nats instance
+    nats_url: String,
+
+    /// NATS client
+    client: Option<Client>,
+}
+
+impl NatsKvStorageEngine {
+    fn new(nats_url: String) -> NatsKvStorageEngine {
+        NatsKvStorageEngine {
+            nats_url,
+            client: None,
+        }
+    }
+}
+
+impl Engine for NatsKvStorageEngine {
+    fn name() -> EngineName {
+        String::from(STORAGE_ENGINE_NAME)
```

Comment:
Nit: in these functions it is better to do `to_owned()` on them. But it is making me think that maybe these functions should return &str

Reply:
I'm trying to prep for a world where the Engine on the other side (maybe I should have made this `async` to show that) could be completely remote (and thus we couldn't have the version as a `&str` per-say). i.e. `RemoteStorageEngine`.

Reply:
That is an interesting use case. Probably keep it for now